### PR TITLE
fix(Core/Events): Corrections to Apothecary Hummel

### PR DESCRIPTION
**Pull request by Nefertumm**
## Changes Proposed:
-  Removed old EventMap implementation
-  Added missing phase.
-  Despawn Crazed Apothecary adds on reset.

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/3037

## Tests Performed:

-  Built but not tested.

## How to Test the Changes:

1. Start event love is in the air.
2. Go to Apothecary Hummel and wipe.
3. It should restart properly.

### CODE REVIEW

**Comment by temperrr** [comment]
Tested

- [ ] "wiping" when his first apothecary isnt active. You see his apothecaries despawn and he is still aggressive.
image

- [ ] Same thing when wiping when all three are active.

File: src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_apothecary_hummel.cpp [hunk 1]
```diff
@@ -116,14 +118,19 @@ class boss_apothecary_hummel : public CreatureScript
         void EnterEvadeMode() override
         {
             summons.DespawnAll();
+            std::list<Creature*> valentineAdds;
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_apothecary_hummel.cpp:121]: the instance spawns them, no?

if (Hummel)
...
hummel->AI()->JustSummoned(add);
...

Comment (Nyeriah) [reply src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_apothecary_hummel.cpp:121]: that way they'll register as his summons and just despawn on evade without the need for searching grid

**Comment by Nefertumm** [comment]
That should work

**Comment by Lynethris** [comment]
Tested and work ;)

**Comment by porkserver** [comment]
Also confirmed working
Screen Shot 2022-03-16 at 5 35 58 PM
Screen Shot 2022-03-16 at 5 29 53 PM

**Review by acidmanifesto — APPROVED** [review]

**Pull request merged**